libqfieldsync/layer.py

```diff
@@ -339,24 +479,77 @@ def apply(self):
             "QFieldSync/value_map_button_interface_threshold",
             self.value_map_button_interface_threshold,
         )
-        # custom properties does not store the data type, so it is safer to remove boolean custom properties, rather than setting them to the string 'false' (which is boolean `True`)
-        if self.is_geometry_locked:
-            self.layer.setCustomProperty("QFieldSync/is_geometry_locked", True)
-        else:
-            self.layer.removeCustomProperty("QFieldSync/is_geometry_locked")
 
-        if self.is_geometry_locked_expression_active:
+        # clear old, outdated properties
+        self.layer.removeCustomProperty("QFieldSync/is_geometry_locked")
+        self.layer.removeCustomProperty(
+            "QFieldSync/is_geometry_locked_expression_active"
+        )
+        self.layer.removeCustomProperty("QFieldSync/geometry_locked_expression")
+
+        if self.is_feature_addition_locked:
+            self.layer.setCustomProperty("QFieldSync/is_feature_addition_locked", True)
+        else:
+            self.layer.removeCustomProperty("QFieldSync/is_feature_addition_locked")
+        if self.is_feature_addition_locked_expression_active:
             self.layer.setCustomProperty(
-                "QFieldSync/is_geometry_locked_expression_active", True
+                "QFieldSync/is_feature_addition_locked_expression_active", True
             )
         else:
             self.layer.removeCustomProperty(
-                "QFieldSync/is_geometry_locked_expression_active"
+                "QFieldSync/is_feature_addition_locked_expression_active"
+            )
+        self.layer.setCustomProperty(
+            "QFieldSync/feature_addition_locked_expression",
+            self.feature_addition_locked_expression,
+        )
+        if self.is_attribute_editing_locked:
+            self.layer.setCustomProperty("QFieldSync/is_attribute_editing_locked", True)
+        else:
+            self.layer.removeCustomProperty("QFieldSync/is_attribute_editing_locked")
```

Comment:
```suggestion
        self.layer.setCustomProperty("QFieldSync/is_attribute_editing_locked", self.is_attribute_editing_locked)
```
Isn't it much more cleaner like this? What is the benefit if we delete the value from the xml?